File: examples/go/grafana-agent-overview/overview.go
package main

import (
	"github.com/grafana/grafana-foundation-sdk/go/common"
	"github.com/grafana/grafana-foundation-sdk/go/dashboard"
	"github.com/grafana/grafana-foundation-sdk/go/table"
	"github.com/grafana/grafana-foundation-sdk/go/units"
)

func runningInstancesTable() *table.PanelBuilder {
	return table.NewPanelBuilder().
		Title("Running Instances").
		Description("General statistics of running grafana agent instances.").
		Height(7).
		Span(24).
		Footer(common.NewTableFooterOptionsBuilder().CountRows(false).Reducer([]string{"sum"})).
		Datasource(datasourceRef("$prometheus_datasource")).
		WithTarget(
			tablePrometheusQuery(`count by (instance, version) (agent_build_info{job=~"$job", instance=~"$instance"})`, "A"),
		).
		WithTarget(
			tablePrometheusQuery(`max by (instance) (time() - process_start_time_seconds{job=~"$job", instance=~"$instance"})`, "B"),
		).
		// Transformations
		WithTransformation(dashboard.DataTransformerConfig{
			Id:      "merge",
			Options: map[string]any{},
		}).
		WithTransformation(dashboard.DataTransformerConfig{
			Id: "organize",
			Options: map[string]any{
				"excludeByName": map[string]any{
					"Time":     true,
					"Value #A": true,
				},
				"renameByName": map[string]any{
					"Value #B": "Uptime",
				},
			},
		}).
		// Overrides
		OverrideByName("Value #B", []dashboard.DynamicConfigValue{
			{Id: "unit", Value: units.Seconds},
		})
}

File: examples/typescript/grafana-agent-overview/src/overview.ts
import * as common from '@grafana/grafana-foundation-sdk/common';
import * as table from "@grafana/grafana-foundation-sdk/table";
import { tablePrometheusQuery } from "./common";

export const runningInstancesTable = (): table.PanelBuilder => {
  return new table.PanelBuilder()
    .title("Running Instances")
    .description("General statistics of running grafana agent instances.")
    .height(7)
    .span(24)
    .footer(
      new common.TableFooterOptionsBuilder()
        .countRows(false)
        .reducer(["sum"])
    )
    .datasource({ uid: "$prometheus_datasource" })
    .withTarget(
      tablePrometheusQuery('count by (instance, version) (agent_build_info{job=~"$job", instance=~"$instance"})', "A"),
    )
    .withTarget(
      tablePrometheusQuery('max by (instance) (time() - process_start_time_seconds{job=~"$job", instance=~"$instance"})', "B"),
    )
    // Transformations
    .withTransformation({
      id: "merge",
      options: {}
    })
    .withTransformation({
      id: "organize",
      options: {
        excludeByName: {
          "Time": true,
          "Value #A": true,
        },
        "renameByName": {
          "Value #B": "Uptime",
        },
      },
    })
    // Overrides
    .overrideByName("Value #B", [
      { id: "unit", value: "s" },
    ]);
};
